src/adserver.js

```diff
@@ -0,0 +1,57 @@
+import {formatQS} from './url';
+
+//Adserver parent class
+var adserver = function(attr) {
+  var that = {};
+  that.name = attr.adserver;
+  that.code = attr.code;
+  that.getWinningBidByCode = function() {
+    var bidObject = $$PREBID_GLOBAL$$._bidsReceived.find(bid => bid.adUnitCode === that.code);
+    return bidObject;
+  };
+  return that;
+};
+
+//DFP ad server
+exports.dfpAdserver = function (options, urlComponents) {
+  var that = adserver(options);
```

Comment:
If you change the `adserver` function as noted in the other comment, you can call it here with `new`. In that case I'd suggest changing this line to something like `var adserver = new AdServer(options);`, and replacing all of the other`that` variables in this function with `adserver`